libs/barista-components/sunburst/src/sunburst.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Overlay, OverlayConfig, OverlayRef } from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnChanges,
+  Optional,
+  Output,
+  SimpleChanges,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  dtSetUiTestAttribute,
+  DtUiTestConfiguration,
+  DT_UI_TEST_CONFIG,
+} from '@dynatrace/barista-components/core';
+import { DtSunburstOverlay } from './sunburst.directive';
+import {
+  DtSunburstNode,
+  DtSunburstNodeInternal,
+  DtSunburstSlice,
+  DtSunburstValueMode,
+  fillNodes,
+  getNodesWithState,
+  getSelectedId,
+  getSelectedNodes,
+  getSelectedNodesFromOutside,
+  getSlices,
+  getValue,
+} from './sunburst.util';
+
+const OVERLAY_PANEL_CLASS = 'dt-sunburst-overlay-panel';
+
+@Component({
+  selector: 'dt-sunburst',
+  templateUrl: 'sunburst.html',
+  styleUrls: ['sunburst.scss'],
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtSunburst implements OnChanges {
+  readonly width = 480;
+  readonly viewBox = '-240 -176 480 352';
+
+  @ViewChild('svg') svgEl;
+
+  /** @internal Template reference for the DtSunburst overlay. */
+  @ContentChild(DtSunburstOverlay, { read: TemplateRef })
+  // tslint:disable-next-line: no-any
+  private _overlay: TemplateRef<any>;
+
+  filledSeries: DtSunburstNodeInternal[];
+  slices: DtSunburstSlice[];
+
+  _selected: DtSunburstNodeInternal[];
+  selectedLabel: string;
+  selectedValue: number;
+  selectedRelativeValue: number;
+  labelAsAbsolute: boolean = true;
+
+  @HostListener('click', ['$event']) onClick(ev?: MouseEvent): void {
+    this.select(ev);
+  }
+
+  @Input() series: DtSunburstNode[];
+  @Input() selected: DtSunburstNode[];
+  @Input() noSelectionLabel: string = 'All';
+  @Input() valueDisplayMode: DtSunburstValueMode = DtSunburstValueMode.ABSOLUTE;
```

Comment:
Please use a type instead of an enum here. I think we have discussed this in #786 
```suggestion
  /** Sets the display mode for the sunburst values to either 'percent' or 'absolute'. 
  @Input() valueDisplayMode: 'percent' | 'absolute' = 'absolute';
```